File: frontend/web/src/types/crm.ts
export type LeadStatus = 'new' | 'contacted' | 'qualified' | 'proposal' | 'won' | 'lost'
export type LeadSource = 'instagram' | 'referral' | 'website' | 'phone' | 'other'

export interface Lead {
  id: string
  name: string
  email?: string
  phone?: string
  company?: string
  status: LeadStatus
  source: LeadSource
  score: number
  notes?: string
  tags: string[]
  created_at: string
  updated_at: string
  last_contact?: string
}

export interface CreateLeadRequest {
  name: string
  email?: string
  phone?: string
  company?: string
  source: LeadSource
  notes?: string
}

export interface LeadFollowUp {
  id: string
  lead_id: string
  message: string
  channel: 'email' | 'whatsapp' | 'phone'
  scheduled_at?: string
  sent_at?: string
  status: 'draft' | 'scheduled' | 'sent'
}

File: backend/app/schemas/crm.py
from pydantic import BaseModel
from datetime import datetime
from enum import Enum

class LeadStatus(str, Enum):
    NEW = "new"
    CONTACTED = "contacted"
    QUALIFIED = "qualified"
    LOST = "lost"
   

class LeadCreate(BaseModel):
    name: str
    email: str
    phone: str | None = None
    status: LeadStatus = LeadStatus.NEW
    source: str | None = None
    notes: str | None = None
    

class LeadResponse(BaseModel):
    id: str
    name: str
    email: str
    phone: str | None = None
    status: LeadStatus
    created_at: datetime

class LeadUpdate(BaseModel):
    name: str | None = None
    email: str | None = None
    phone: str | None = None
    status: LeadStatus | None = None